factor/email/tests/factor_test.php

```diff
@@ -100,12 +100,12 @@ public function test_generate_email_ip_address_location(string $ip, bool $expect
         $unknownstr = get_string('email:geoinfo:unknown', 'factor_email');
 
         if ($expectedunknown) {
-            $this->assertContains($unknownstr, $email);
+            $this->assertStringContainsString($unknownstr, $email);
```

Comment:
We'll want to check if this function exists first before using it. IIRC I purposely did not use assertStringContainsString because it caused the CI to fail because it didn't exist in the CI's phpunit version.

Reply:
Or another option to avoid having an ugly if else check if to use assertNotEquals(0, strpos(....)